test/Microsoft.Data.Analysis.Tests/DataFrameTests.IDataView.cs

```diff
@@ -224,5 +224,59 @@ public void TestIDataViewWithNulls()
             Assert.Equal("", preview.ColumnView[14].Values[5].ToString()); // null row
             Assert.Equal("foo", preview.ColumnView[14].Values[6].ToString());
         }
+
+        [Fact]
+        public void TestDataFrameFromIDataView()
+        {
+            DataFrame df = DataFrameTests.MakeDataFrameWithAllColumnTypes(10, withNulls: false);
+            df.Columns.Remove("Char"); // Because chars are returned as uint16 by IDataView, so end up comparing CharDataFrameColumn to UInt16DataFrameColumn and fail asserts
+            IDataView dfAsIDataView = df;
+            DataFrame newDf = dfAsIDataView.ToDataFrame();
+            Assert.Equal(dfAsIDataView.GetRowCount(), newDf.Rows.Count);
+            Assert.Equal(dfAsIDataView.Schema.Count, newDf.Columns.Count);
+            for (int i = 0; i < df.Columns.Count; i++)
+            {
+                Assert.True(df.Columns[i].ElementwiseEquals(newDf.Columns[i]).All());
+            }
+        }
+
+        [Fact]
+        public void TestDataFrameFromIDataView_SelectColumns()
+        {
+            DataFrame df = DataFrameTests.MakeDataFrameWithAllColumnTypes(10, withNulls: false);
+            IDataView dfAsIDataView = df;
+            DataFrame newDf = dfAsIDataView.ToDataFrame("Int", "Double");
+            Assert.Equal(dfAsIDataView.GetRowCount(), newDf.Rows.Count);
+            Assert.Equal(2, newDf.Columns.Count);
+            Assert.True(df.Columns["Int"].ElementwiseEquals(newDf.Columns["Int"]).All());
+            Assert.True(df.Columns["Double"].ElementwiseEquals(newDf.Columns["Double"]).All());
+        }
+
+        [Fact]
+        public void TestDataFrameFromIDataView_SelectRows()
+        {
+            DataFrame df = DataFrameTests.MakeDataFrameWithAllColumnTypes(10, withNulls: false);
+            df.Columns.Remove("Char"); // Because chars are returned as uint16 by IDataView, so end up comparing CharDataFrameColumn to UInt16DataFrameColumn and fail asserts
+            IDataView dfAsIDataView = df;
+            DataFrame newDf = dfAsIDataView.ToDataFrame(5);
+            Assert.Equal(5, newDf.Rows.Count);
+            Assert.Equal(dfAsIDataView.Schema.Count, newDf.Columns.Count);
+            for (int i = 0; i < df.Columns.Count; i++)
+            {
+                Assert.True(df.Columns[i].ElementwiseEquals(newDf.Columns[i]).All());
+            }
+        }
+
+        [Fact]
+        public void TestDataFrameFromIDataView_SelectColumnsAndRows()
+        {
+            DataFrame df = DataFrameTests.MakeDataFrameWithAllColumnTypes(10, withNulls: false);
```

Comment:
It may make sense to get an ML.NET `IDataView` (maybe read a text file or possibly something that has been predicted against) and use that. That way we are testing against something other than `DataFrame`'s IDataView implementation.

Reply:
You could also use `mlContext.Data.LoadFromEnumerable` if you wanted to define it all in code.

Reply:
Yup, `LoadFromEnumerable` looks promising. Will add a test based on that and push everything tomorrow